src/auth/auth.service.ts

```diff
@@ -37,13 +49,19 @@ export class AuthService {
       throw new UnauthorizedException();
     }
 
-    return this.jwtService.signAsync(
-      {
-        id: user.id,
-        email: user.email,
-      },
-      { expiresIn: `${process.env.JWT_EXPIRY_TIME}s` },
-    );
+    if (user.enabled2FA) {
+      return this.jwtService.signAsync(
+        {
+          id: user.id,
+          status: 'FURTHER_ACTION_IS_REQUIRED',
+          message:
+            'Go to http://localhost:3000/auth/verify-2fa to continue authentication',
```

Comment:
`http://localhost:3000` part should be taken from environment variables. Otherwise it won't work on environments different from development / local one.

Reply:
seeing that this is a token payload not a HTTP response, I'm not sure if `status` and `message` are necessary - I would remove them from the token payload I guess